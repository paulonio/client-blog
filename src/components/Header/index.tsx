import React from 'react';

import Navigation from '@/components/Navigation';

import { ROUTES } from '@/constants/routes';

import styles from './styled.module.scss';

const Header = () => {
  return (
    <header className={styles.header}>
      <div className={`container ${styles.header__wrapper}`}>
        <h1 className={styles.title}>Modsen Client Blog</h1>
        <div className={styles.right}>
          <nav>
            <Navigation links={ROUTES} />
          </nav>
          <button type="button" className={styles.button}>
            Video about us
          </button>
        </div>
      </div>
    </header>
  );
};

export default Header;
