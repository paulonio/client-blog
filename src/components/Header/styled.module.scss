@import '../../styles/sass/constants';

.header {
  background-color: $color-black;

  &__wrapper {
    height: 80px;
    display: flex;
    align-items: center;
    justify-content: space-between;
  }
}

.title {
  font-size: 24px;
  line-height: 48px;
  font-weight: 700;
  letter-spacing: 0;
  color: $color-white;
}

.right {
  display: flex;
  align-items: center;
}

.navigation {
  display: flex;
  align-items: center;

  &__link {
    margin-left: 24px;
    list-style: none;
    &:first-child {
      margin-left: 0;
    }

    & a {
      font-size: 16px;
      line-height: 28px;
      font-weight: 400;
      transition: 0.3s;
      &:hover {
        color: $color-primary-hover;
      }
    }
  }
}

.button {
  margin-left: 32px;
  padding: 16px 48px;
  font-size: 18px;
  line-height: 24px;
  font-weight: 700;
  color: $color-black;
  background-color: $color-white;
  border: none;
  transition: background-color 0.3s;
  &:hover {
    background-color: $color-light-s;
    cursor: pointer;
  }
}