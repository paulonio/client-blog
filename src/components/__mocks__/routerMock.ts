import { useRouter } from 'next/router';

jest.mock('next/router', () => ({
  useRouter: jest.fn(),
}));

(useRouter as jest.Mock).mockReturnValue({ query: {} });
