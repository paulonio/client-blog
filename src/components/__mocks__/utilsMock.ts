import { isLinkActive, parseString } from '@/utils/utils';

jest.mock('../../utils/utils', () => ({
  isLinkActive: jest.fn(),
  parseString: jest.fn(),
}));

(isLinkActive as jest.Mock).mockReturnValue(true);
(parseString as jest.Mock).mockReturnValue('');
