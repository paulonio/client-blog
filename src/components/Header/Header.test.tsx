import React from 'react';
import { render } from '@testing-library/react';

import '../__mocks__/routerMock';

import Header from '.';

describe('Header', () => {
  it('should render header', () => {
    const { container } = render(<Header />);

    expect(container).toMatchSnapshot();
  });
});
