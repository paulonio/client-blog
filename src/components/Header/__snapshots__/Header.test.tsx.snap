// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should render header 1`] = `
<div>
  <header
    class="header"
  >
    <div
      class="container header__wrapper"
    >
      <h1
        class="title"
      >
        title
      </h1>
      <div
        class="right"
      >
        <nav>
          <ul
            class="navigation"
          >
            <li
              class="navigation__link className"
            >
              <a
                href="/"
              >
                home
              </a>
            </li>
            <li
              class="navigation__link className"
            >
              <a
                href="/blog"
              >
                blog
              </a>
            </li>
            <li
              class="navigation__link className"
            >
              <a
                href="/about"
              >
                about-us
              </a>
            </li>
            <li
              class="navigation__link className"
            >
              <a
                href="/contact"
              >
                contact-us
              </a>
            </li>
            <li
              class="navigation__link className"
            >
              <a
                href="/"
              >
                change
              </a>
            </li>
          </ul>
        </nav>
        <button
          class="button"
          type="button"
        >
          video
        </button>
        <div
          aria-hidden="true"
          class="overlay "
        />
        <div
          aria-hidden="true"
          class="wrapper "
        >
          <div
            class="line"
          />
        </div>
      </div>
    </div>
  </header>
</div>
`;
